indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -305,6 +346,125 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
   }
 
   /**
+   * Check priorities of existing tasks, if all has lesser priorities and no lock is non-preemptive then return true
+   * If this task can acquire lock and some lower priority TaskLocks need to be revoked then populate tasksToPreempt Map with the entry - taskId and TaskLock
+   * */
+  private Pair<Boolean, Map<String, TaskLock>> canAcquireLock(final List<TaskLockPosse> foundPosses, int priority) {
+    boolean canAcquire = true;
+    final Map<String, TaskLock> tasksToPreempt = new HashMap<>();
+    for (TaskLockPosse taskLockPosse : foundPosses) {
+      if(priority <= taskLockPosse.getTaskLock().getPriority() || !taskLockPosse.getTaskLock().isPreemptive()) {
+        // Check whether the tasks (in the taskLockPosse) which are blocking the new task are actually running,
```

Comment:
This check was introduced to protect against zombie tasklocks where there is no associated task with an active lock. It was a while back when we observed this problem where we recovered our metadata store from a snapshot. Unfortunately, at that time we didn't had enough time to reproduce this problem, therefore added this check as a defensive mechanism. I can remove the check for now and use active tasks. If the problem happens again in future then I can add this check again. What do you think ?